test/API/H5_api_dataset_test.c

```diff
@@ -4464,426 +4452,1516 @@ test_dataset_io_point_selections(void)
 
-        /* Fill write buffer */
-        for (i = 0; i < DATASET_IO_POINT_DIM_0; i++)
-            for (j = 0; j < DATASET_IO_POINT_DIM_1; j++)
-                buf_all[i][j] = rand();
+    for (i = 0; i < DATASET_MULTI_COUNT; i++) {
+        char dset_name[DSET_NAME_BUF_SIZE];
 
-        /* Write data */
-        if (H5Dwrite(dset_id, H5T_NATIVE_INT, H5S_ALL, H5S_ALL, H5P_DEFAULT, buf_all) < 0)
-            FAIL_PUTS_ERROR("Failed to write entire dataset");
+        if (DSET_NAME_BUF_SIZE <= snprintf(dset_name, DSET_NAME_BUF_SIZE, "%s%zu%c",
+                                           DATASET_SMALL_READ_TEST_ALL_DSET_NAME, i, '\0'))
+            TEST_ERROR;
 
-        /* Update file_state */
-        for (i = 0; i < DATASET_IO_POINT_DIM_0; i++)
-            for (j = 0; j < DATASET_IO_POINT_DIM_1; j++)
-                file_state[i][j] = buf_all[i][j];
+        if ((dset_id_arr[i] = H5Dcreate2(group_id, dset_name, DATASET_SMALL_READ_TEST_ALL_DSET_DTYPE,
+                                         fspace_id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT)) < 0) {
+            H5_FAILED();
+            printf("    couldn't create dataset '%s'\n", DATASET_SMALL_READ_TEST_ALL_DSET_NAME);
+            goto error;
+        }
 
-        /* Generate points to read */
-        DATASET_IO_POINT_GEN_POINTS(points, i, j);
+        fspace_id_arr[i] = fspace_id;
+        dtype_id_arr[i]  = DATASET_SMALL_READ_TEST_ALL_DSET_DTYPE;
 
-        /* Select points */
-        if (H5Sselect_elements(fspace_id, H5S_SELECT_SET, DATASET_IO_POINT_NPOINTS, points) < 0)
+        if (NULL == (read_buf_arr[i] = malloc(data_size)))
             TEST_ERROR;
+    }
 
-        /* Wipe read buffer */
-        memset(buf_point, 0, sizeof(buf_point));
-
-        /* Read points to "all" memory buffer */
-        if (H5Dread(dset_id, H5T_NATIVE_INT, mspace_id_all, fspace_id, H5P_DEFAULT, buf_point) < 0)
-            FAIL_PUTS_ERROR("Failed to read points from dataset to all memory buffer");
+    // TODO
+    fprintf(stderr, "H5Dcreate complete");
 
-        /* Verify data */
-        for (i = 0; i < DATASET_IO_POINT_NPOINTS; i++)
-            if (buf_point[i] != file_state[points[2 * i]][points[2 * i + 1]])
-                FAIL_PUTS_ERROR("Incorrect data read from points to all memory buffer");
+    if (H5Dread_multi(DATASET_MULTI_COUNT, dset_id_arr, dtype_id_arr, fspace_id_arr, fspace_id_arr,
+                      H5P_DEFAULT, read_buf_arr) < 0) {
+        H5_FAILED();
+        printf("    couldn't read from dataset '%s'\n", DATASET_SMALL_READ_TEST_ALL_DSET_NAME);
+        goto error;
+    }
 
-        /* Generate points to write */
-        DATASET_IO_POINT_GEN_POINTS(points, i, j);
+    // TODO
+    fprintf(stderr, "H5Dread_multi complete");
 
-        /* Select points */
-        if (H5Sselect_elements(fspace_id, H5S_SELECT_SET, DATASET_IO_POINT_NPOINTS, points) < 0)
+    for (i = 0; i < DATASET_MULTI_COUNT; i++)
+        free(read_buf_arr[i]);
```

Comment:
should set these to NULL after freeing each one so that if an error happens after this point, the test doesn't crash from a double free